src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs

```diff
@@ -425,69 +452,132 @@ public static CultureInfo CurrentCulture
                     throw new ArgumentNullException(nameof(value));
                 }
 
-#if ENABLE_WINRT
-                WinRTInteropCallbacks callbacks = WinRTInterop.UnsafeCallbacks;
-                if (callbacks != null && callbacks.IsAppxModel())
+#if FEATURE_APPX
+                if (ApplicationModel.IsUap)
                 {
-                    callbacks.SetGlobalDefaultCulture(value);
-                    return;
+                    if (SetCultureInfoForUserPreferredLanguageInAppX(value))
+                    {
+                        // successfully set the culture, otherwise fallback to legacy path
+                        return; 
+                    }
                 }
 #endif
-
-                s_currentThreadCulture = value;
+                if (s_asyncLocalCurrentCulture == null)
+                {
+                    Interlocked.CompareExchange(ref s_asyncLocalCurrentCulture, new AsyncLocal<CultureInfo>(AsyncLocalSetCurrentCulture), null);
+                }
+                s_asyncLocalCurrentCulture.Value = value;
             }
         }
 
         public static CultureInfo CurrentUICulture
         {
             get
             {
-                CultureInfo ci = GetUserDefaultCultureCacheOverride();
-                if (ci != null)
+#if FEATURE_APPX
+                if (ApplicationModel.IsUap)
                 {
-                    return ci;
+                    CultureInfo culture = GetCultureInfoForUserPreferredLanguageInAppX();
+                    if (culture != null)
+                        return culture;
                 }
+#endif
```

Comment:
This one is present in CoreCLR and not in CoreRT (same as in CurrentCulture), need to investigate. In fact the presence of this code is the same reason why GetCurrentUICultureNoAppX is needed.